### DISCUSSION TITLE
Git-crypt does not work for Flatpak installation

### DISCUSSION THREAD

**Discussion by gperens** [Vinzent03/obsidian-git#289] [Problems]
Any Git command that uses git-crypt will fail with error. Works fine with Appimage version. I have a feeling that it is similar sandbox problem like "Snap puts Obsidian in kind of a sandbox, so that Obsidian can't access Git", but with git-crypt.
If there is a fix or workaround, I would be interested since I do prefer Flatpak to Appimage.

## Environment
- Pop!_OS 22.04 LTS Linux
- Obsidian 0.15.9 Flatpak installation
- Obsidian git 1.31.0
- Git-crypt 0.7.0

## Debug info: 
```
git obsidian error: "git-crypt" clean: line 1: git-crypt: command not found
error: external filter '"git-crypt" clean' failed 127
error: external filter '"git-crypt" clean' failed
fatal: test/test.md: clean filter 'crypt' failed
```

**Comment by Vinzent03** [comment]
Is git-crypt installed in the same location as git? Should be `/usr/bin`

**Reply by gperens** [reply]
Yes, it is at /usr/bin/git-crypt

**Reply by Vinzent03** [reply]
Can you open the console `CTRL + SHIFT + I` and enter in both versions `process.env["PATH"]` into the console? It should be the same, but who knows.

**Reply by gperens** [reply]
By both versions I assume you mean Flatpak and Appimage, so I tried it.
Flatpak:
```
> process.env["PATH"]
'/app/bin:/usr/bin:/app/bin:/app/texlive/bin:/app/texlive/bin/x86_64-linux:/app/texlive/bin/aarch64-linux'
```
Appimage:
``` 
> process.env["PATH"]
'/tmp/.mount_ObsidiWbSVQP:/tmp/.mount_ObsidiWbSVQP/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin'
```

**Reply by gperens** [reply]
@Vinzent03 Anything else I could try to solve this?

**Reply by Vinzent03** [reply]
Did you try specifying the absolute path as custom binary path? So not just the executable name, but the whole path.

**Reply by gperens** [reply]
@Vinzent03 Thanks for the reply! Sorry, but I am not sure what you mean by "specifying the absolute path as custom binary path" as to where should I specify this. What I tried during my troubleshooting, is to change the `clean` command filter in `.git/config` to absolute path of git-crypt or to change the format, but none of it worked.

**Reply by Vinzent03** [reply]
I guess you are currently setting `git-crypt` as custom binary path. You may try using the absolute path of `git-crypt`. E.g. `/usr/bin/git-crypt`, but I don't know where it's stored.

**Reply by Vinzent03** [reply]
Does this issue help you? https://github.com/denolehov/obsidian-git/issues/250

**Reply by gperens** [reply]
Issue #250 was the one that I followed to try to change the config file.
Once `git-crypt init ` is ran in any local Git repo, it automatically changes the file `.git/config` and adds the following filter:
```
[filter "git-crypt"]
	smudge = \"git-crypt\" smudge
	clean = \"git-crypt\" clean
	required = true
```
So I tried to change that `clean` line (because that is the command which fails) by replacing it with absolute path:
`clean = \"/usr/bin/git-crypt\" clean` but that did not work or maybe I did not get the syntax right (use of quotes is little weird here).

**Comment by zufanka** [comment]
Hi! I have the same problem in Flatpak

```
git obsidian error: "/usr/bin/git-crypt" clean: line 1: /usr/bin/git-crypt: No such file or directory
error: external filter '"/usr/bin/git-crypt" clean' failed 127
error: external filter '"/usr/bin/git-crypt" clean' failed
```

**Reply by gperens** [reply]
And I still have not found a fix. Currently using Appimage, but would love to go back to Flatpak if this issue would be solved.

**Reply by Vinzent03** [reply]
I think you have to create an issue on the Flatpak side https://github.com/flathub/md.obsidian.Obsidian/issues
